pkg/driver/controller_test.go

```diff
@@ -47,107 +47,6 @@ const (
 	expDevicePath = "/dev/xvda"
 )
 
-func TestNewControllerService(t *testing.T) {
```

Comment:
Because this unit test is a disaster 😄 

To give a bit more context here, the previous implementation relied on global variables to override their assignment for the single purpose of unit testing:

```
var (
	// NewMetadataFunc is a variable for the cloud.NewMetadata function that can
	// be overwritten in unit tests.
	NewMetadataFunc = cloud.NewMetadataService
	// NewCloudFunc is a variable for the cloud.NewCloud function that can
	// be overwritten in unit tests.
	NewCloudFunc = cloud.NewCloud
)
```

metadata and cloud are created within `newControllerService` which makes unit testing its functionality incredibly difficult. 

Fixing the relationship between cloud, controller, and metadata is beyond the scope of this PR, but I will be addressing that in an upcoming PR to fix this pain point. For now, I've rolled back metadata instantiation in controller `cloud.NewMetadataService(cfg, region)` -> `NewMetadataFunc(cfg, region)` to re-implement unit test.